lib/galaxy/dependencies/lint-requirements.txt

```diff
@@ -0,0 +1,3 @@
+flake8
```

Comment:
We've learned from experience we want to pin these right - otherwise we will have newer checks break older versions of Galaxy?

Reply:
Leaving these unpinned sometimes broke the CI lint jobs, yes.
I'm struggling to find a way to:
- don't have to install all dev requirements (which is a superset of the normal requirements) to run `tox -e py36-lint`
- automate flake8* pinning updates
- don't break CI lint jobs when flake8 or a plugin change the rules

all at the same time. Suggestions welcome!

Reply:
I haven't seen anything in poetry to help :(. It is just three dependencies currently - we could just pin them at their current versions and manually modify that file?

That can still break if dependencies of dependencies break, a slightly more robust solution is we could keep it unpinned in dev and then build something into the release process to pip install that file and replace it with a locked version with ``pip freeze``.

Reply:
I was also thinking about `pip freeze`, I'll give it a shot!
BTW, we could potentially have the same issue with mypy, should I add it to the lint requirements?

Reply:
Yeah - I'm sure you're right. That would be nice and very much appreciated.

Reply:
@jmchilton Rebased with the requested changes (pinned lint requirements, including mypy).